src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActor.scala

```diff
@@ -65,7 +66,17 @@ private[impl] class KillServiceActor(
     }
   }
 
+  def initializeWithDecommissionedInstances() = async {
+    val toKillBasedOnGoal = await(instanceTracker.instancesBySpec())
+      .allInstances
+      .filter(i => i.state.goal != Goal.Running && i.isActive)
+
+    self ! KillInstancesAndForget(toKillBasedOnGoal)
+  }
```

Comment:
Just some nitpicking: 

```scala
instanceTracker.instancesBySpec().map { bySpec =>
  bySpec.allInstances.filter(...)
}.pipeTo(self)
```

`async { ... }` returns a future which is confusing since we don't want to wait for it. Also, any exception in the block is lot unless you pipe it. 

```scala
async {
  await(instanceTracker.instancesBySpec())
  .allInstances
  .filter(...)
}.pipeTo(self)
```

would be fine as well.

Reply:
Fixed